tests/kvlds/main.c

```diff
@@ -183,11 +183,11 @@ add(struct wire_requestqueue * Q,
 	}
 
 	/* Check for no-op. */
-	if (noop && op_p) {
+	if (noop && !op_p) {
```

Comment:
Instead of changing all these values, you might prefer it if we negated `done` when setting `op_p` on line 69.

Reply:
Yes.  The value `op_p` means "did we perform an operation".  The function `callback_donep` should take a final parameter of `int status` and set `op_p = status ? 0 : 1`.